tests/datasets/test_metrics_processor.py

```diff
@@ -106,7 +108,7 @@
                 "tags.key": [10, 20, 30],
                 "tags.value": [11, 22, 33],
                 "values": [324.12, 345.23, 4564.56, 567567],
-                "materialization_version": 2,
+                "materialization_version": MATERIALIZATION_VERSION,
```

Comment:
Do you plan to add a test for the new processor ?

Reply:
We are in effect doing this in the e2e test_metrics_api.py because we're using the polymorphic bucket as the incoming writable storage, but I can write some unit tests more specific to the processor itself.

Reply:
I don't think we should rely on the end to end tests only ([test pyramid](https://martinfowler.com/articles/practical-test-pyramid.html)). Everything is covered there, this is true, but that hardly help you in testing a functionality. You cannot test corner cases and if something break, you will just see a large test failing without having any hint on what broke (query path, consumer, processor, etc.).
The unit test for the processor still has the value to validate that the processor is processing the messages appropriately. It only covers that area, it can cover corner cases and if that breaks you know the processor is broken, no need for additional investigation

Reply:
yeah, in between my response and your reply I actually added unit tests for the polymorphic processor